Sources/ZcashLightClientKit/Synchronizer/SDKSynchronizer.swift

```diff
@@ -652,6 +652,173 @@ public class SDKSynchronizer: Synchronizer {
+        }
+        
+        // Initialize services for the endpoints
+        let services = endpoints.map {
+            Service(
+                originalEndpoint: $0,
+                service: LightWalletGRPCService(
+                    host: $0.host,
+                    port: $0.port,
+                    secure: $0.secure,
+                    singleCallTimeout: 5000,
+                    streamingCallTimeout: Int64(fetchThresholdSeconds) * 1000
+                ),
+                url: "\($0.host):\($0.port)"
+            )
+        }
+
+        // Parallel part
+        var checkResults: [String: CheckResult] = [:]
+        
+        await withTaskGroup(of: CheckResult.self) { group in
+            for service in services {
+                group.addTask {
+                    let startTime = Date().timeIntervalSince1970
+                    let info = try? await service.service.getInfo()
+                    let markTime = Date().timeIntervalSince1970
+                    let latestBlockHeight = try? await service.service.latestBlockHeight()
+                    let endTime = Date().timeIntervalSince1970
+
+                    let getInfoTime = markTime - startTime
+                    let latestBlockHeightTime = endTime - markTime
+                    let mean = (getInfoTime + latestBlockHeightTime) / 2
+
+                    return CheckResult(
+                        id: service.url,
+                        info: info,
+                        getInfoTime: getInfoTime,
+                        latestBlockHeight: latestBlockHeight,
+                        latestBlockHeightTime: latestBlockHeightTime,
+                        mean: mean,
+                        service: service,
+                        blockTime: 0
+                    )
+                }
+            }
+            
+            var tmpResults: [String: CheckResult] = [:]
+            
+            for await result in group {
+                // rule out results where calls failed
+                guard let info = result.info, let _ = result.latestBlockHeight else {
+                    continue
+                }
+                
+                // rule out if mismatch of networks
+                guard (info.chainName == "main" && network == .mainnet) 
+                    || (info.chainName == "test" && network == .testnet) else {
+                    continue
+                }
+                
+                // rule out mismatch of consensus branch IDs
+                guard let localBranchID = await blockProcessor.consensusBranchIdFor(Int32(info.blockHeight)) else {
+                    continue
+                }
+
+                guard let remoteBranchID = ConsensusBranchID.fromString(info.consensusBranchID) else {
+                    continue
+                }
+                
+                guard remoteBranchID == localBranchID else {
+                    continue
+                }
+
+                // rule out syncing server
```

Comment:
```suggestion
                // Rule out servers that are syncing, stuck, or probably on the wrong fork.
                // To avoid falsely ruling out all servers this can only be a very loose check
                // (i.e. `nBlocks` should not be too small), because `info.estimatedHeight`
                // may be quite inaccurate.
```
(Replace `nBlocks` with the new name if it is renamed.)

Reply:
Updated